src/engine/permission-handler.ts

```diff
@@ -0,0 +1,86 @@
+export interface Permissions {
+  [to: string]: {
+    [from: string]: {
+      allow: boolean,
+      hash: string,
+    }
+  }
+}
+
+export interface IPermissionProvider {
+  confirm(message: string): Promise<{ allow: boolean, remember: boolean }>
+  /** Clear the permission object */
+  clear(): void
+}
+
+export interface IPermissionHandler {
+  permissions: Permissions
+  askPermission(from: {name: string, hash: string}, to: string): Promise<boolean>
+}
+
+/**
+ * Example of a PermissionHandler using localStorage
+ */
+export abstract class PermissionHandler implements IPermissionHandler, IPermissionProvider {
+
+  public permissions: Permissions
+  abstract confirm(message: string): Promise<{ allow: boolean, remember: boolean }>
+
+  constructor() {
+    this.permissions = JSON.parse(localStorage.getItem('permissions'))
+  }
+
+  private setPermissions() {
+    const permissions = JSON.stringify(this.permissions)
+    localStorage.setItem('permissions', permissions)
+  }
+
+  public clear() {
+    localStorage.removeItem('permissions')
+  }
+
+  /**
+   * Show a message to ask the user for a permission
+   * @param from The name and hash of the plugin that make the call
+   * @param to The name of the plugin that receive the call
+   * @param wasAllow Did the plugin have changed its hash
+   */
+  private async openPermission(
+    from: {name: string, hash: string},
+    to: string, wasAllow: boolean
+  ): Promise<{ allow: boolean, remember: boolean }> {
+    const msg = wasAllow
+      ? `${from.name} would like to access plugin ${to}.\n Check it's new content ${from.hash}`
+      : `${from.name} has changed and would like to access the plugin ${to}.\n Check it's new content ${from.hash}`
+    return this.confirm(msg)
```

Comment:
could we pass in profile of both plugin / module ? so we can display more information if needed?

Reply:
Sure